test/jit/test_pdt.py

```diff
@@ -0,0 +1,189 @@
+import os
+import sys
+import torch
+from torch.testing._internal.jit_utils import JitTestCase, make_global
+from torch.jit._monkeytype_config import _IS_MONKEYTYPE_INSTALLED
+from typing import List, Dict, Tuple  # noqa: F401
+
+# Make the helper files in test/ importable
+pytorch_test_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+sys.path.append(pytorch_test_dir)
+
+if not _IS_MONKEYTYPE_INSTALLED:
+    print("monkeytype is not installed. Skipping tests for Profile-Directed Typing", file=sys.stderr)
+    JitTestCase = object  # type: ignore[misc, assignment] # noqa: F811
+
+if __name__ == "__main__":
+    raise RuntimeError(
+        "This test file is not meant to be run directly, use:\n\n"
+        "\tpython test/test_jit.py TESTNAME\n\n"
+        "instead."
+    )
+
+class TestPDT(JitTestCase):
+    """
+    A suite of tests for profile directed typing in TorchScript.
+    """
+    def test_pdt(self):
+        def test_sum(a, b):
+            return a + b
+
+        make_global(test_sum)
```

Comment:
I think we don't need `make_global` for functions, just classes.

Reply:
I tried removing global for tests, it did not work.

Reply:
I remember now why `make_global` is required. `monkeytype` actually fails to recognize the `test_sum` inside `test_pdt`. It only recognizes when it is made global. 

Here is the error:
```
Traceback (most recent call last):
  File "/data/users/nikithamalgi/pytorch/test/jit/test_pdt.py", line 100, in test_pdt
    self.assertEqual(scripted_fn_add(10, 2), test_sum(10, 2))
  File "/data/users/nikithamalgi/pytorch/torch/testing/_internal/common_utils.py", line 133, in prof_func_call
    return prof_callable(func_call, *args, **kwargs)
  File "/data/users/nikithamalgi/pytorch/torch/testing/_internal/common_utils.py", line 130, in prof_callable
    return callable(*args, **kwargs)
RuntimeError: test_sum() Expected a value of type 'Tensor (inferred)' for argument 'a' but instead found type 'int'.
Inferred 'a' to be of type 'Tensor' because it was not annotated with an explicit type.
Position: 0
Value: 10
Declaration: test_sum(Tensor a, Tensor b) -> (Tensor)
Cast error details: Unable to cast Python instance to C++ type (compile in debug mode for details)
```

Reply:
I see. Is this a fundamental limitation of MonkeyType? ie, MonkeyType can not profile nested functions?

We don't need to do anything about it in this PR, but if this is a limitation of MonkeyType, we would need keep it in mind.